src/ray/rpc/node_manager/node_manager_client_pool.h

```diff
@@ -40,6 +40,7 @@ class NodeManagerClientPool {
 
   /// Return an existing NodeManagerWorkerClient if exists, and connect to one if it does
   /// not. The returned pointer is borrowed, and expected to be used briefly.
+  /// The function is guaranteed to return the non-nullptr.
```

Comment:
For these type of in or out values, you could use `absl::Nonnull` for clarify :)